src/app/results/results.component.ts

```diff
@@ -170,22 +173,38 @@ export class ResultsComponent implements OnInit {
       this.presentPage = Math.abs(query['start'] / this.searchdata.rows) + 1;
 
     });
+
     this.totalResults$.subscribe(totalResults => {
       if (totalResults) {
         this.hidefooter = 0;
-
       }
 
       this.end = Math.min(totalResults, this.begin + this.searchdata.rows - 1);
       this.message =  'showing results ' + this.begin + ' to ' + this.end + ' of ' + totalResults;
       this.noOfPages = Math.ceil(totalResults / this.searchdata.rows);
       this.maxPage = Math.min(this.searchdata.rows, this.noOfPages);
     });
-
-
   };
 
   ngOnInit() {
 
   }
+
+  filterLink(link) {
```

Comment:
excluding results should not be done on front end as the count of results per page decreases. In whatever situation we have to display 10 results per page. 
Okay to be clear,
suppose we have 10 initial results and all are jpg, this gives us zero results on first page,on which user assumes that there are no results supporting his search query.